control/controlclient/map.go

```diff
@@ -693,6 +693,21 @@ func peerChangeDiff(was tailcfg.NodeView, n *tailcfg.Node) (_ *tailcfg.PeerChang
 			if va == nil || vb == nil || *va != *vb {
 				return nil, false
 			}
+		case "ExitNodeDNSResolvers":
+			va, vb := was.ExitNodeDNSResolvers(), views.SliceOfViews(n.ExitNodeDNSResolvers)
+
+			if va.Len() != vb.Len() {
+				return nil, false
+			}
+
+			for i := range va.LenIter() {
+				for j := range vb.LenIter() {
+					if !va.At(i).Equal(vb.At(j)) {
+						return nil, false
+					}
+				}
+			}
```

Comment:
I think this results in calling the inner `for j := range vb.LenIter()` loop once per entry in `va`? I think this does the same, since you check the length above:
```suggestion
			for i := range va.LenIter() {
				if !va.At(i).Equal(vb.At(j)) {
					return nil, false
				}
			}
```